cordova-lib/src/plugman/platforms/browser.js

```diff
@@ -31,25 +29,7 @@ module.exports = {
         return path.join(project_dir, 'www');
     },
     package_name:function(project_dir) {
-        // preferred location if cordova >= 3.4
-        var preferred_path = path.join(project_dir, 'config.xml');
-        var config_path;
-
-        if (!fs.existsSync(preferred_path)) {
-            // older location
-            var old_config_path = path.join(module.exports.www_dir(project_dir), 'config.xml');
-            if (!fs.existsSync(old_config_path)) {
-                // output newer location and fail reading
-                config_path = preferred_path;
-                events.emit('verbose', 'unable to find '+config_path);
-            } else {
-                config_path = old_config_path;
-            }
-        } else {
-            config_path = preferred_path;
-        }
-        var widget_doc = xml_helpers.parseElementtreeSync(config_path);
-        return widget_doc._root.attrib['id'];
+        return common.package_name(project_dir);

```

Comment:
Agree about `firefoxos` and `webos` (I'd add here `tizen` as well)

As for `blackberry10` - it does not rely on configs other than the one at the platform `www` dir, so using common `package_name` might lead to unexpected results. Also seems that it doesn't affected by this bug as it always copying own `config.xml` at `create`